python_modules/dagster/dagster_tests/execution_tests/test_job.py

```diff
@@ -196,3 +196,35 @@ def test_job_input_values_out_of_process():
     with instance_for_test() as instance:
         result = execute_pipeline(reconstructable(pass_from_job), instance=instance)
         assert result.success
+
+
+def test_subset_job_with_config():
+    @op
+    def echo(x: int):
+        return x
+
+    @job
+    def no_config():
+        echo(echo.alias("emit")())
+
+    result = no_config.execute_in_process(run_config={"ops": {"emit": {"inputs": {"x": 1}}}})
+    assert result.success
+
+    subset_no_config = no_config.get_job_def_for_subset_selection(op_selection=["echo"])
+
+    result = subset_no_config.execute_in_process(run_config={"ops": {"echo": {"inputs": {"x": 1}}}})
+    assert result.success
+
+    @job(config={"ops": {"emit": {"inputs": {"x": 1}}}})
+    def with_config():
+        echo(echo.alias("emit")())
+
+    result = with_config.execute_in_process()
+    assert result.success
+
+    subset_with_config = with_config.get_job_def_for_subset_selection(op_selection=["echo"])
+
+    result = subset_with_config.execute_in_process(
```

Comment:
Would this test have failed previously if run_config was left blank here? (my understanding is "no", but perhaps I'm missing something)

I think there is a fair scope of area where you'd be ok with the old behavior. Basically, in any case where you're passing in actual op config rather than input config, and you only have config defined for a small number of ops in your graph, most graph subselections that include all of the configured ops should work.

Reply:
> Would this test have failed previously if run_config was left blank here?

it would fail because the subset `echo` requires input config now since its inputs have been disconnected